### DISCUSSION TITLE
Can semicolon node separators be used across multiple lines?

### DISCUSSION THREAD

**Discussion by Devasta** [kdl-org/kdl#179] [Q&A] [answered]
Hello!

I intend to make an XSL transform that when complete will hopefully allow you turn any XML 1.0 document into an XiK file.

Just a question, if you don't mind:

I can see that the spec says:

> Finally, a node is terminated by either a Newline, a Children Block, a semicolon (;) or the end of the file/stream (an EOF).

```
stooges {
    Larry
    Curly
    Moe
}

stooges {
    Larry; Curly; Moe;
}
```

So both of the above would be valid. My question is, are you allowed have both newlines and semicolons? Specifically, is the below allowed?

```
stooges {
    Larry;
    Curly;
    Moe;
}
```

Many thanks in advance.

Regards

Dan

**Comment by zkat** [comment, marked as answer]
Yes, that's perfectly legal because nodes can be preceded by newlines and whitespace. You just can't do multiple semicolons

**Reply by Devasta** [reply]
Thank you for the reply, appreciate it! :)
